### PR TITLE
git-clone: 0.8 -> 0.9, bumping git-init image

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

This bumps the git-init image to 0.40.0 instead of 0.29.0.

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations][authoring]
- [x] Includes [docs][docs] (if user facing)
- [x] Includes [tests][tests] (for new tasks or changed functionality)
  - See the [end-to-end testing documentation][e2e] for guidance and CI details.
- [x] Meets the [Tekton contributor standards][contributor] (including functionality, content, code)
- [x] Commit messages follow [commit message best practices][commit]
- [x] Has a kind label. You can add one by adding a comment on this PR that
  contains `/kind <type>`. Valid types are bug, cleanup, design, documentation,
  feature, flake, misc, question, tep
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md) for more details._

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]: https://github.com/tektoncd/catalog/issues/413
[authoring]: https://github.com/tektoncd/catalog/blob/main/recommendations.md
[docs]: https://github.com/tektoncd/community/blob/master/standards.md#docs
[tests]: https://github.com/tektoncd/community/blob/master/standards.md#tests
[e2e]: https://github.com/tektoncd/catalog/blob/main/CONTRIBUTING.md#end-to-end-testing
[contributor]: https://github.com/tektoncd/community/blob/main/standards.md
[commit]: https://github.com/tektoncd/community/blob/master/standards.md#commit-messages

### CODE REVIEW

**Comment by chmouel** [comment]
weird error that seems relevant,

it's been so long that it wasnt update that nobody noticed therre was new features needed to be updated  in relation with git-init binary changes 😅

**Comment by chmouel** [comment]
spare checkout works when launching directly so not sure what is it 

```
% ./bin/git-init '-url=https://github.com/kelseyhightower/nocode' '-revision=' '-refspec=' '-path=/tmp/output/' '-sslVerify=true' '-submodules=true' '-depth=1' '-sparseCheckoutDirectories=CONTRIBUTING.md,STYLE.md' -terminationMessagePath=/tmp/foobar
{"level":"info","ts":1663671691.9207299,"caller":"git/git.go:176","msg":"Successfully cloned https://github.com/kelseyhightower/nocode @ 6c073b08f7987018cbb2cb9a5747c84913b3608e (grafted, HEAD) in path /tmp/output/"}
{"level":"info","ts":1663671691.9672074,"caller":"git/git.go:215","msg":"Successfully initialized and updated submodules in path /tmp/output/"}
```

File: task/git-clone/0.9/git-clone.yaml [hunk 1]
```diff
@@ -0,0 +1,238 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: git-clone
+  labels:
+    app.kubernetes.io/version: "0.8"
```
Comment (vinamra28) [inline task/git-clone/0.9/git-clone.yaml:6]:
```suggestion
    app.kubernetes.io/version: "0.9"
```

File: task/git-clone/0.9/git-clone.yaml [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+      type: string
+      default: ""
+    - name: verbose
+      description: Log the commands that are executed during `git-clone`'s operation.
+      type: string
+      default: "true"
+    - name: gitInitImage
+      description: The image providing the git-init binary that this Task runs.
+      type: string
+      default: "gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/git-init:v0.40.0"
```
Comment (vinamra28) [inline task/git-clone/0.9/git-clone.yaml:105]: is this image backward compatible, atleast with pipelines version 0.38.0 ?

File: task/git-clone/0.9/git-clone.yaml [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+        value: $(workspaces.basic-auth.bound)
+      - name: WORKSPACE_BASIC_AUTH_DIRECTORY_PATH
+        value: $(workspaces.basic-auth.path)
+      - name: WORKSPACE_SSL_CA_DIRECTORY_BOUND
+        value: $(workspaces.ssl-ca-directory.bound)
+      - name: WORKSPACE_SSL_CA_DIRECTORY_PATH
+        value: $(workspaces.ssl-ca-directory.path)
+      securityContext:
+        runAsNonRoot: true
+        runAsUser: 65532
```
Comment (vinamra28) [inline task/git-clone/0.9/git-clone.yaml:168]: I guess this is the reason of why tests are failing

**Comment by vinamra28** [comment]
@vdemeester any updates on this? 😅

**Comment by vdemeester** [comment]
Closing as https://github.com/tektoncd/catalog/pull/1092 has been merged.

**Pull request closed without merging**